server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAOBuilder.java

```diff
@@ -68,11 +72,22 @@ public ReadyToBuild payloadCodec(Optional<PayloadCodec> payloadCodec) {
             return this;
         }
 
+        public ReadyToBuild putBlob(Optional<Function<Blob, BlobId>> putBlob) {
+            this.putBlob = putBlob;
+            return this;
+        }
+
         public ObjectStorageBlobsDAO build() {
             Preconditions.checkState(containerName != null);
             Preconditions.checkState(blobIdFactory != null);
 
-            return new ObjectStorageBlobsDAO(containerName, blobIdFactory, supplier.get(), payloadCodec.orElse(PayloadCodec.DEFAULT_CODEC));
+            BlobStore blobStore = supplier.get();
+
+            return new ObjectStorageBlobsDAO(containerName,
+                blobIdFactory,
+                blobStore,
+                putBlob.orElse((blob) -> blobIdFactory.from(blobStore.putBlob(containerName.value(), blob))),
```

Comment:
you could write a method named `defaultPutBlob` that is referenced here instead?